### PR TITLE
allow multiple pipelines being fetched per ref

### PR DESCRIPTION
**Pull request by calmez**
this allows multiple metrics on one ref to be published

### CODE REVIEW

File: pkg/controller/pipelines.go [hunk 1]
```diff
@@ -42,10 +34,11 @@ func (c *Controller) PullRefMetrics(ctx context.Context, ref schemas.Ref) error
 	}
 
 	pipelines, _, err := c.Gitlab.GetProjectPipelines(ctx, ref.Project.Name, &goGitlab.ListProjectPipelinesOptions{
-		// We only need the most recent pipeline
 		ListOptions: goGitlab.ListOptions{
-			PerPage: 1,
+			PerPage: int(ref.Project.Pull.Pipeline.PerRef),
 			Page:    1,
+			OrderBy: "updated_at",
```
Comment (spitfjre) [inline pkg/controller/pipelines.go]: These two are new. Should this be removed or made configurable?

File: pkg/store/local.go [hunk 2]
```diff
@@ -21,6 +21,9 @@ type Local struct {
 	metrics      schemas.Metrics
 	metricsMutex sync.RWMutex
 
+	pipelineVariables       map[float64]string
+	pipelineVariablesMutext sync.RWMutex
```
Comment (spitfjre) [inline pkg/store/local.go]: Mutext -> Mutex

File: pkg/controller/pipelines.go [hunk 3]
```diff
@@ -59,22 +52,57 @@ func (c *Controller) PullRefMetrics(ctx context.Context, ref schemas.Ref) error
 		return nil
 	}
 
-	pipeline, err := c.Gitlab.GetRefPipeline(ctx, ref, pipelines[0].ID)
+	slices.Reverse(pipelines)
```
Comment (goir) [inline pkg/controller/pipelines.go:55]: why is this reversed ?

**Review by goir — APPROVED** [review]

File: pkg/store/local.go (hunk 2 above)
Comment (calmez) [reply pkg/store/local.go]: Thanks for catching that.

File: pkg/controller/pipelines.go (hunk 1 above)
Comment (calmez) [reply pkg/controller/pipelines.go]: True, removed it.

File: pkg/controller/pipelines.go (hunk 3 above)
Comment (calmez) [reply pkg/controller/pipelines.go:55]: This is also to make the looping code as backwards compatible to the vanilla version of the code as possible.
The `ProcessPipelineMetrics` method has the side effect to store a `LatestPipeline` in the `ref`. By reversing we're processing the first one last giving us the same `LatestPipeline` as the vanilla code. This is more a precaution to not break any assumptions about the `LatestPipeline`. Other than that the order of the processed pipelines should not matter all that much.
Maybe I can leave a comment explaining this in the code. After a week I also needed a hard think why that was needed.

Comment (calmez) [reply pkg/controller/pipelines.go:55]: The tests would be green even without this line, but as I said I leave it more as a precaution to not break assumptions that the original author has on the `ref.LatestPipeline`.

**Pull request merged**